### PR TITLE
Refactor/kws 1799 get random user

### PR DESCRIPTION
**Pull request by GuilhermeMota93**

### CODE REVIEW

File: kwssdk/src/main/java/kws/superawesome/tv/kwssdk/base/providers/GetRandomUsernameProvider.kt [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+            // network success case
+            if (rawString != null && networkError == null) {
+
+                val parseRequest = ParseJsonRequest(rawString = rawString)
+                val parseTask = ParseJsonTask()
+                val authResponse = parseTask.execute<AppConfigResponse>(input = parseRequest,
+                        clazz = AppConfigResponse::class.java) ?: AppConfigResponse()
+
+                val appConfigAppResponseObj = authResponse.appConfigAppObject
```
Comment (devgabrielcoman) [inline kwssdk/src/main/java/kws/superawesome/tv/kwssdk/base/providers/GetRandomUsernameProvider.kt]: lines 65 to 73 can actually be replace by this:

```
val appConfigAppResponseObj = authResponse.appConfigAppObject
val error = if (appConfigAppResponseObj != null) null else Throwable("Error - not found valid app config")
callback(appConfigAppResponseObj, error)
```

File: kwssdk/src/main/java/kws/superawesome/tv/kwssdk/base/providers/GetRandomUsernameProvider.kt [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        val networkGetRandomUsernameRequest = GetRandomUsernameRequest(
+                environment = environment,
+                appID = id)
+
+        val networkGetRandomUsernameTask = NetworkTask()
+        networkGetRandomUsernameTask.execute(input = networkGetRandomUsernameRequest) { rawString, networkError ->
+
+            if (rawString != null && networkError == null) {
+
+                val randomUserName = rawString.replace("\"", "")
```
Comment (devgabrielcoman) [inline kwssdk/src/main/java/kws/superawesome/tv/kwssdk/base/providers/GetRandomUsernameProvider.kt]: What does happen when **rawString.replace** fails?

File: kwssdk/src/main/java/kws/superawesome/tv/kwssdk/base/requests/GetAppConfigRequest.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+    override val endpoint: String = "v1/apps/config"
+
+    override val query: Map<String, Any>? = mapOf("oauthClientId" to clientID)
+
+    override val method: NetworkMethod = NetworkMethod.GET
+
+    override val body: Map<String, Any>? = null
+
+    override val formEncodeUrls: Boolean
```
Comment (devgabrielcoman) [inline kwssdk/src/main/java/kws/superawesome/tv/kwssdk/base/requests/GetAppConfigRequest.kt]: can remove this line, it defaults to **false** anyway

File: kwssdk/src/main/java/kws/superawesome/tv/kwssdk/base/requests/GetRandomUsernameRequest.kt [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+internal class GetRandomUsernameRequest(environment: KWSNetworkEnvironment, appID: Int)
+    : BaseRequest(environment = environment) {
+
+    override val headers: Map<String, String> = mapOf("Content-Type" to "application/json")
+
+    override val endpoint: String = "v2/apps/$appID/random-display-name"
+
+    override val method: NetworkMethod = NetworkMethod.GET
+
+    override val body: Map<String, Any>? = null
```
Comment (devgabrielcoman) [inline kwssdk/src/main/java/kws/superawesome/tv/kwssdk/base/requests/GetRandomUsernameRequest.kt]: should remove the **body** as well, it defaults to null anyways

File: kwssdk/src/main/java/kws/superawesome/tv/kwssdk/base/providers/GetRandomUsernameProvider.kt (hunk 2 above)
Comment (GuilhermeMota93) [reply kwssdk/src/main/java/kws/superawesome/tv/kwssdk/base/providers/GetRandomUsernameProvider.kt]: Added null check validation

**Pull request merged**